functions/issue-scraper.ts

```diff
@@ -0,0 +1,351 @@
+    return new Response("Internal Server Error", {
+      headers: corsHeaders,
+      status: 500,
+    });
+  }
+}
+
+function markdownToPlainText(markdown: string | null): string | null {
+  if (!markdown) return markdown;
+  const md = markdownit() as MarkdownItWithPlainText;
+  md.use(plainTextPlugin);
+  md.render(markdown);
+  return md.plainText;
+}
+
+const SEARCH_ISSUES_QUERY = `
+  query SearchIssues($searchText: String!, $after: String) {
+    search(
+      query: $searchText,
+      type: ISSUE,
+      first: 100,
+      after: $after
+    ) {
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      nodes {
+        ... on Issue {
+          id
+          number
+          title
+          body
+          state
+          stateReason
+          createdAt
+          updatedAt
+          closedAt
+          author {
+            login
+          }
+          assignees(first: 10) {
+            nodes {
+              login
+            }
+          }
+          repository {
+            id
+            name
+            owner {
+              login
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+async function fetchUserIssuesBatch(octokit: InstanceType<typeof Octokit>, username: string, lastScraped?: number): Promise<IssueNode[]> {
+  const allIssues: IssueNode[] = [];
+  let hasNextPage = true;
+  let cursor: string | null = null;
+
+  // Construct the query with the lastScraped timestamp
+  const searchText = `assignee:${username} is:issue is:closed reason:completed ${lastScraped ? `closed:>${new Date(lastScraped).toISOString()}` : ""}`;
+
+  while (hasNextPage) {
+    const variables: { searchText: string; after?: string } = { searchText };
+    if (cursor) {
+      variables.after = cursor;
+    }
+
+    const response: GraphQlSearchResponse = await octokit.graphql<GraphQlSearchResponse>(SEARCH_ISSUES_QUERY, variables);
+
+    allIssues.push(...response.search.nodes);
+
+    hasNextPage = response.search.pageInfo.hasNextPage;
+    cursor = response.search.pageInfo.endCursor;
+  }
+
+  return allIssues;
+}
+
+async function batchEmbeddings(voyageClient: VoyageAIClient, texts: string[]): Promise<(number[] | undefined)[]> {
+  try {
+    const embeddingResponse = await voyageClient.embed({
+      input: texts,
+      model: "voyage-large-2-instruct",
+      inputType: "document",
+    });
+    return embeddingResponse.data?.map((item) => item.embedding) || [];
+  } catch (error) {
+    console.error("Error batching embeddings:", error);
+    throw error;
+  }
+}
+
+async function batchUpsertIssues(
+  supabase: SupabaseClient,
+  issues: Array<{
+    id: string;
+    markdown: string;
+    plaintext: string;
+    embedding: string;
+    author_id: number;
+    payload: PayloadType;
+  }>
+): Promise<void> {
+  const { error } = await supabase.from("issues").upsert(issues);
+  if (error) {
+    throw new Error(`Error during batch upsert: ${error.message}`);
```

Comment:
Perhaps we can try an exponential backoff? I wonder if this is worth implementing to make this more robust although I imagine that errors are quite rare for this.

Reply:
Retries should be handled from the client so no timeouts on the worker side should be relevant.

Reply:
That's what I would implement if I was working on this code yeah